matchmake-extension/database/join_matchmake_session.go

```diff
@@ -0,0 +1,41 @@
+package database
+
+import (
+	"github.com/PretendoNetwork/nex-go/v2"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/v2/globals"
+	match_making_database "github.com/PretendoNetwork/nex-protocols-common-go/v2/match-making/database"
+	"github.com/PretendoNetwork/nex-protocols-common-go/v2/matchmake-extension/tracking"
+	match_making_types "github.com/PretendoNetwork/nex-protocols-go/v2/match-making/types"
+)
+
+// JoinMatchmakeSession joins participants from the same connection into a MatchmakeSession. Returns the new number of participants
+func JoinMatchmakeSession(manager *common_globals.MatchmakingManager, matchmakeSession match_making_types.MatchmakeSession, connection *nex.PRUDPConnection, vacantParticipants uint16, joinMessage string) (uint32, *nex.Error) {
+	newParticipants, nexError := match_making_database.JoinGathering(manager, uint32(matchmakeSession.ID), connection, vacantParticipants, joinMessage)
+	if nexError != nil {
+		return 0, nexError
+	}
+
+	// TODO - Should we return the error in these cases?
+	if matchmakeSession.MatchmakeSystemType == 5 { // * Attached to a persistent gathering
```

Comment:
Same comment about the constants, we know the real names of these https://github.com/tech-ticks/RTDXTools/blob/232a7797e01369e9c12704f58fdde65dd3ac1c32/Assets/Scripts/Stubs/Generated/Assembly-CSharp/NexPlugin/Matchmake.cs#L48-L52